# 14.2 try-catch-finally中return前finally是否执行



## 题目描述

在Java中，若在try-catch-finally的try语句块中插入return语句，finally语句块是否还会执行？



## 解答

是的，finally会执行。当退出try语句块时，finally语句块将会执行。即使我试图从try语句块里跳出（通过return语句、continue、break或异常抛出），finally语句块仍将得到执行。

除非以下情况：

1）如果虚拟机在try-catch语句块执行期间退出；

2）如果执行try-catch语句块的线程被杀死终止了